editor/filesystem_dock.cpp

```diff
@@ -2797,6 +2891,27 @@ void FileSystemDock::_file_and_folders_fill_popup(PopupMenu *p_popup, Vector<Str
 		}
 
 		p_popup->add_separator();
+
+		if (p_paths[0] != "res://") {
+			PopupMenu *folder_colors_menu = memnew(PopupMenu);
```

Comment:
This isn't correct either, the menu is not freed automatically, so you are just making infinite menus (and only the first one is used), see #79965

Reply:
Would it be sufficient for now to create the folder colors submenu in the constructor & parent it to the tree PopupMenu initially, but `reparent()` it when it needs to be moved?

Reply:
The PR I linked got approved and might be merged soon, so it will resolve this issue. It's probably fine to leave it as is right now, sorry for bother >_>

Reply:
No worries haha, that works.